web/src/filter.js

```diff
@@ -1030,43 +1030,26 @@ export class Filter {
         return "";
     }
 
-    static describe_is_operator(operator) {
-        const verb = operator.negated ? "exclude " : "";
-        const operand = operator.operand;
-
-        switch (operand) {
-            case "starred":
-            case "alerted":
-            case "unread":
-                return verb + operand + " messages";
-            case "mentioned":
-                return verb + "@-mentions";
-            case "dm":
-            case "private":
-                return verb + "direct messages";
-            case "resolved":
-                return verb + "topics marked as resolved";
-        }
-
-        return "invalid " + operand + " operand for is operator";
-    }
-
     // Convert a list of operators to a human-readable description.
-    static describe_unescaped(operators) {
+    static parts_for_describe(operators) {
+        const parts = [];
+
         if (operators.length === 0) {
-            return "all messages";
+            parts.push({type: "plain_text", content: "all messages"});
+            return parts;
         }
```

Comment:
We should move `describe_unescaped` part of the change to a separate commit.

Reply:
Can you explain how you're thinking about separating them? I see this function change as creating the data needed for the template, which would be necessary to happen at the same time as the rest of this commit.

Reply:
I'm not sure I see how to do a good separation here.